### PR TITLE
Rename CClaimTrieCache to CClaimTrieUpdateBuffer and separate in own file

### PR DESCRIPTION
**Pull request by bvbfan**
Signed-off-by: Anthony Fieroni <bvbfan@abv.bg>

### CODE REVIEW

**Comment by ekxoiner** [comment]
How to get start claim lbry coin?

**Comment by bvbfan** [comment]
This a code review can you ask here https://www.facebook.com/lbryio

**Comment by kaykurokawa** [comment]
CClaimTrie.nCurrentHeight has also been renamed to nNextHeight here

File: src/test/claimtrieupdatebuffer_tests.cpp [hunk 1]
```diff
@@ -67,18 +68,18 @@ BOOST_AUTO_TEST_CASE(recursiveprune_test)
     base_node.insertClaim(test_claim);
 
     // node 1 has a claim so it should not be pruned
-    CClaimTrieNode node_1;
+    CClaimTrieNode *node_1 = new CClaimTrieNode;
```
Comment (BrannonKing) [inline src/test/claimtrieupdatebuffer_tests.cpp]: The old code looked more correct to me. Why the change to an allocation? The modern c++ says that the `new` keyword is generally deprecated. Use `std/boost::make_unique` or `make_shared` if you really need to allocate on the heap, but try to avoid it generally.

Comment (bvbfan) [reply src/test/claimtrieupdatebuffer_tests.cpp]: Children are vector of shared pointers

Comment (bvbfan) [reply src/test/claimtrieupdatebuffer_tests.cpp]: new is completely not deprecated or so on.

**Comment by kaykurokawa** [comment]
I had some problem rebasing this, there is basically a commit here:
7d321ec0eac0a9cb14a3ef078f134a87f6aed956 which looks like a failed merge commit that contained elements from unrelated commits. 

I've resolved this in  https://github.com/lbryio/lbrycrd/pull/175 so will be closing this one. 

In the future, I suggest that you rebase commits that you are working on top of master instead of merging it.
